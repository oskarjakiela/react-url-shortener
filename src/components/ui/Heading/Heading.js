import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import './Heading.css';


const displayName = 'Shorty-Heading';

const Heading = ({
  children,
  className,
  ...props
}) => (
  <h2 className={classnames(displayName, className)} {...props}>
    {children}
  </h2>
);

Heading.displayName = displayName;

Heading.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Heading;
