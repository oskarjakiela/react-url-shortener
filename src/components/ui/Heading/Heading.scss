.Shorty-Heading {
  margin-top: 0;
  margin-bottom: 0;
  font-family: 'Roboto', sans-serif;
  font-weight: 300;
  font-size: 1.375rem;
  color: #555;
}
