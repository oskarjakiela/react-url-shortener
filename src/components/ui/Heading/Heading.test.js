import React from 'react';
import renderer from 'react-test-renderer';

import Heading from './Heading';


it('<Heading /> renders correctly', () => {
  const tree = renderer
    .create(<Heading>Lorem ipsum</Heading>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Heading className="foo-bar" /> renders correctly', () => {
  const tree = renderer
    .create(<Heading className="foo-bar">Lorem ipsum</Heading>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Heading id="foo-bar" /> renders correctly', () => {
  const tree = renderer
    .create(<Heading id="foo-bar">Lorem ipsum</Heading>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
