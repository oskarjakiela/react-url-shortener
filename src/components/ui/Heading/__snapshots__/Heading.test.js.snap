// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> renders correctly 1`] = `
<h2
  className="Shorty-Heading"
>
  Lorem ipsum
</h2>
`;

exports[`<Heading className="foo-bar" /> renders correctly 1`] = `
<h2
  className="Shorty-Heading foo-bar"
>
  Lorem ipsum
</h2>
`;

exports[`<Heading id="foo-bar" /> renders correctly 1`] = `
<h2
  className="Shorty-Heading"
  id="foo-bar"
>
  Lorem ipsum
</h2>
`;
